data_processors/pipeline/domain/workflow.py

```diff
@@ -54,6 +57,75 @@ class Helper(object):
     pass
 
 
+class EngineParameter(ABC):
+    """
+    Abstract EngineParameter that provide methods for generating _dynamic_ engine parameter values
+    instead of loading them as template from SSM Parameter Store.
```

Comment:
Question: engineParameters are always dynamic? That means we'd get rid of the static TF defined SSM parameter/template?

I guess that will work for this case, but the SSM parameter template was meant for parameters/settings that are fixed and stable for all workflows (of the same type). I don't think we have a use case for that right now (i.e. all our engineParameters *have to* be dynamically modified for each workflow execution). So I am fine with a completely dynamic creation *instead* of loading a template.
I just wonder if there is/could be a benefit in loading a template from SSM and filling in the dynamic bits. (also because that would mean one place to look for workflow inputs/config)

Reply:
> That means we'd get rid of the static TF defined SSM parameter/template?

Yes

> engineParameters are always dynamic? I just wonder if there is/could be a benefit in loading a template from SSM and filling in the dynamic bits.

Yep; agree and figure that too. Same dilemma. Just thought I'd do refactor/deprecating about that with next PR. May be I should do right away, I reckon.

Reply:
Refactor is a tad easier than thought. It is fixed with b2a8116